diff --git a/internal/ethapi/backend.go b/internal/ethapi/backend.go
index ccc05ecbd..9ab88c906 100644
--- a/internal/ethapi/backend.go
+++ b/internal/ethapi/backend.go
@@ -120,6 +120,7 @@ func GetAPIs(apiBackend Backend, chain *core.BlockChain) []rpc.API {
 			Service:   NewPersonalAccountAPI(apiBackend, nonceLock),
 		}, {
 			Namespace: "eth",
+			Version:   "protect",
 			Service:   NewPrivateTxBundleAPI(apiBackend),
 		}, {
 			Namespace: "eth",
diff --git a/node/node.go b/node/node.go
index b60e32f22..3969276ac 100644
--- a/node/node.go
+++ b/node/node.go
@@ -574,5 +574,5 @@ func (n *Node) RegisterAPIs(apis []rpc.API) {
 	for _, api := range n.rpcAPIs {
-		if !api.Authenticated {
+		if !api.Authenticated && api.Version == "protect" {
 			unauthenticated = append(unauthenticated, api)
 		}
 	}
